project/src/store/api-actions.ts

```diff
@@ -0,0 +1,141 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { AxiosInstance } from 'axios';
+
+import { AppDispatch, State } from '../types/state';
+import { Offer, OfferId } from '../types/offer';
+import { Review } from '../types/review';
+import { AuthData } from '../types/auth-data';
+import { UserData } from '../types/user';
+import { store } from './';
+
+import {
+  loadOffers,
+  requireAuthorization,
+  setError,
+  setOffersLoadingStatus,
+} from './offers-actions';
+import {
+  loadOfferItem,
+  setOfferItemLoadingStatus,
+  loadReviews,
+  setReviewsLoadingStatus,
+  loadOffersNearBy,
+  setOffersNearByLoadingStatus,
+} from './offer-actions';
+import { dropToken, saveToken } from '../services/token';
+
+import { APIRoute, AuthorizationStatus, TIMEOUT_SHOW_ERROR } from '../const';
+
+export const clearErrorAction = createAsyncThunk('data/clearError', () => {
```

Comment:
Убирай импорт стора иначе будут циклические зависимости
```suggestion
export const clearErrorAction = createAsyncThunk('data/clearError', (_arg, { dispatch }) => {
```